controller_frequency: 10.0
recovery_behaviour_enabled: true
initial_stage_number: 1
max_stage_number: 2
collision_check_frequency: 10
smooth_tolerance: 0.01
smooth_delta: 0.5
smooth_weight: 0.5

NavfnROS:
  allow_unknown: true # Specifies whether or not to allow navfn to create plans that traverse unknown space.
  default_tolerance: 0.1 # A tolerance on the goal point for the planner.

TrajectoryPlannerROS:
  # Robot Configuration Parameters
  acc_lim_x: 2.5
  acc_lim_theta:  3.2

  max_vel_x: 1.0
  min_vel_x: 0.0

  max_vel_theta: 1.0
  min_vel_theta: -1.0
  min_in_place_vel_theta: 0.2

  holonomic_robot: false
  escape_vel: -0.1

  # Goal Tolerance Parameters
  yaw_goal_tolerance: 0.1
  xy_goal_tolerance: 0.2
  latch_xy_goal_tolerance: false

  # Forward Simulation Parameters
  sim_time: 2.0
  sim_granularity: 0.02
  angular_sim_granularity: 0.02
  vx_samples: 6
  vtheta_samples: 20
  controller_frequency: 20.0

  # Trajectory scoring parameters
  meter_scoring: true # Whether the gdist_scale and pdist_scale parameters should assume that goal_distance and path_distance are expressed in units of meters or cells. Cells are assumed by default (false).
  occdist_scale:  0.1 #The weighting for how much the controller should attempt to avoid obstacles. default 0.01
  pdist_scale: 0.75  #     The weighting for how much the controller should stay close to the path it was given . default 0.6
  gdist_scale: 1.0 #     The weighting for how much the controller should attempt to reach its local goal, also controls speed  default 0.8

  heading_lookahead: 0.325  #How far to look ahead in meters when scoring different in-place-rotation trajectories
  heading_scoring: false  #Whether to score based on the robot's heading to the path or its distance from the path. default false
  heading_scoring_timestep: 0.8   #How far to look ahead in time in seconds along the simulated trajectory when using heading scoring (double, default: 0.8)
  dwa: true #Whether to use the Dynamic Window Approach (DWA)_ or whether to use Trajectory Rollout
  simple_attractor: false
  publish_cost_grid_pc: true  

  # Oscillation Prevention Parameters
  oscillation_reset_dist: 0.25 #How far the robot must travel in meters before oscillation flags are reset (double, default: 0.05)
  escape_reset_dist: 0.1
  escape_reset_theta: 0.1

DWAPlannerROS:
  # Robot configuration parameters  
  acc_lim_x: 2.5
  acc_lim_y: 0
  acc_lim_th: 3.2

  max_vel_x: 3.0
  min_vel_x: -1.0
  max_vel_y: 0
  min_vel_y: 0

  max_trans_vel: 2.0
  min_trans_vel: 1.5
  max_rot_vel: 1.0
  min_rot_vel: 0.4  
  acc_lim_x: 2.5
  acc_lim_theta:  3.2

  # Goal Tolerance Parameters
  yaw_goal_tolerance: 0.05
  xy_goal_tolerance: 0.1
  latch_xy_goal_tolerance: false
  use_dwa: false

  # # Forward Simulation Parameters
  sim_time: 2.0
  # sim_granularity: 0.02
  vx_samples: 6
  vy_samples: 1
  vtheta_samples: 20
  penalize_negative_x: true

  # # Trajectory scoring parameters
  path_distance_bias: 1.0 #36.0 
  goal_distance_bias: 1.0 #24.0
  occdist_scale: 1.0 
  forward_point_distance: 0.01 # 0.325 
  # stop_time_buffer: 0.2 
  # scaling_speed: 0.25 
  # max_scaling_factor: 0.2 

  # # Oscillation Prevention Parameters
  # oscillation_reset_dist: 0.25 

  index_scale: 2.0
  dist_scale: 1.0
  close_index_scale: 1.0
  backward_penalty_scale: 4.0
  length_cost_scale: 0.5
  obstacle_cost_scale: 2.0

middle_planner:
  # odom_topic: odom
  # Robot configuration parameters  
  acc_lim_x: 2.5
  acc_lim_y: 0
  acc_lim_th: 3.2

  max_vel_x: 2.0
  min_vel_x: -2.0
  max_vel_y: 0
  min_vel_y: 0

  max_trans_vel: 2.0
  min_trans_vel: 0.0
  max_rot_vel: 0.5
  min_rot_vel: 0.1

  # Goal Tolerance Parameters
  yaw_goal_tolerance: 0.1
  xy_goal_tolerance: 0.5
  latch_xy_goal_tolerance: false

  # # Forward Simulation Parameters
  sim_time: 4.0
  use_dwa: true
  sim_granularity: 0.02
  vx_samples: 5
  vy_samples: 1
  vth_samples: 10


  # # Trajectory scoring parameters
  path_distance_bias: 1.0 
  goal_distance_bias: 1.0 
  occdist_scale: 1.0 
  forward_point_distance: 1.5 
  # stop_time_buffer: 0.2  
  # scaling_speed: 0.25
  # max_scaling_factor: 0.2 
  index_scale: 2.0
  dist_scale: 3.0
  close_index_scale: 1.0
  backward_penalty_scale: 3.0
  length_cost_scale: 0.5
  obstacle_cost_scale: 4.0

  # # Oscillation Prevention Parameters
  oscillation_reset_dist: 1.0 # 0.25 
